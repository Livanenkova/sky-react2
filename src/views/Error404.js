import { Link} from 'react-router-dom'

export default function Error404() {

  return (
    <div className="error404">
      <h1>error404</h1>
      <Link to="/">Main Page</Link>
    </div>
  )
}