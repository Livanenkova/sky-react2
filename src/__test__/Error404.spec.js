/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React, {Component} from 'react'
import renderer from 'react-test-renderer';
import {Router } from 'react-router-dom';
import ReactTestUtils from 'react-dom/test-utils'; 
import Error404 from '../views/Error404';

const { createMemoryHistory } = require("history");





describe('Error404 Component', () => {
  it('отображается корректно', () => {
    const history = createMemoryHistory();
    const tree = renderer.create(<Router location={history.location} navigator={history}><Error404/></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('has an h1 tag', () => {
  //     const component = ReactTestUtils.renderIntoDocument(<Error404/>);    
  //     let h1 = ReactTestUtils.findRenderedDOMComponentWithTag(
  //     component, 'h1'
  //   )
  // });
  })
