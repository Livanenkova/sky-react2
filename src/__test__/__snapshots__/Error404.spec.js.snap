// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error404 Component отображается корректно 1`] = `
<div
  className="error404"
>
  <h1>
    error404
  </h1>
  <a
    href="/"
    onClick={[Function]}
  >
    Main Page
  </a>
</div>
`;

exports[`отображается корректно 1`] = `
<div
  className="error404"
>
  <h1>
    error404
  </h1>
  <a
    href="/"
    onClick={[Function]}
  >
    Main Page
  </a>
</div>
`;
